diff --git a/dist/index.mjs b/dist/index.mjs
index 4ae3af3677276bf97861d49f010d5d32f654ac5a..9cdc72aff8b0d6ba1d76e14af6d9dcb9644272d6 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -1060,7 +1060,7 @@ function parseToMemberExpressionOrIdentifier(code) {
 function viteQiankun(opts) {
   let config;
   const qiankunWindow = `__QIANKUN_WINDOW__["${opts.name}"]`;
-  let publicPath = `(${qiankunWindow}.__INJECTED_PUBLIC_PATH_BY_QIANKUN__ || "")`;
+  let publicPath = `(window.__INJECTED_PUBLIC_PATH_BY_QIANKUN__ || "")`;
   return [
     {
       name: "qiankun:remain-exports",
